syntax = "proto2";

package Authentication;

message CreateAccountWeb
{
  optional int64 requestId = 1;
  optional string email = 2;
  optional string plaintextPassword = 3;
}

message CreateAccountWebSuccess
{
  optional int64 requestId = 1;
  optional int64 userId = 2;
}

message CreateAccountWebFailure
{
  optional int64 requestId = 1;

  enum Reason 
  {
    ACCOUNT_ALREADY_EXISTS = 1;
    INVALID_PASSWORD = 2;
    INTERNAL_SERVER_ERROR = 3;
  }

  optional Reason reason = 2;
}

message AuthenticateWeb 
{
  optional int64 requestId = 1;
  optional string email = 2;
  optional string plaintextPassword = 3;
}

message AuthenticateWebSuccess 
{
  optional int64 requestId = 1;
  optional int64 userId = 2;
  optional string creationDate = 3;
}

message AuthenticateWebFailure 
{
  optional int64 requestId = 1;

  enum Reason 
  {
    INVALID_CREDENTIALS = 1;
    INTERNAL_SERVER_ERROR = 2;
  }

  optional Reason reason = 2;
}
